hudi-client/hudi-client-common/src/main/java/org/apache/hudi/table/action/commit/BaseWriteHelper.java

```diff
@@ -46,22 +47,31 @@ public HoodieWriteMetadata<O> write(String instantTime,
                                       int configuredShuffleParallelism,
                                       BaseCommitActionExecutor<T, I, K, O, R> executor,
                                       WriteOperationType operationType) {
+    return this.write(instantTime, inputRecords, context, table, shouldCombine, configuredShuffleParallelism, executor, operationType, Option.empty());
+  }
+
+  public HoodieWriteMetadata<O> write(String instantTime,
+                                      I inputRecords,
+                                      HoodieEngineContext context,
+                                      HoodieTable<T, I, K, O> table,
+                                      boolean shouldCombine,
+                                      int configuredShuffleParallelism,
+                                      BaseCommitActionExecutor<T, I, K, O, R> executor,
+                                      WriteOperationType operationType,
+                                      Option<HoodieTimer> sourceReadAndIndexTimer) {
     try {
       // De-dupe/merge if needed
       I dedupedRecords =
           combineOnCondition(shouldCombine, inputRecords, configuredShuffleParallelism, table);
 
-      Instant lookupBegin = Instant.now();
       I taggedRecords = dedupedRecords;
```

Comment:
Same question, why not just initialzie the timer here so that we can avoid to introduce a new method.

Reply:
makes sense